contracts/CoveragePool.sol

```diff
@@ -185,60 +146,63 @@ contract CoveragePool is Ownable {
     {
-        assetPool.approveNewAssetPoolUpgrade(_newAssetPool);
+        assetPool.approveNewAssetPoolUpgrade(newAssetPool);
     }
 
-    /// @notice Returns the time remaining until the risk manager approval
-    ///         process can be finalized
-    /// @param riskManager Risk manager in the process of approval
-    /// @return Remaining time in seconds.
-    function getRemainingRiskManagerApprovalTime(address riskManager)
+    function beginWithdrawalDelayUpdate(uint256 newWithdrawalDelay)
         external
-        view
-        returns (uint256)
+        onlyOwner
     {
-        return
-            getRemainingChangeTime(
-                riskManagerApprovalTimestamps[riskManager],
-                CoveragePoolConstants.RISK_MANAGER_GOVERNANCE_DELAY,
-                "Risk manager approval not initiated"
-            );
+        assetPool.beginWithdrawalDelayUpdate(newWithdrawalDelay);
+    }
+
+    function finalizeWithdrawalDelayUpdate() external onlyOwner {
+        assetPool.finalizeWithdrawalDelayUpdate();
     }
 
-    /// @notice Returns the time remaining until the risk manager unapproval
+    function beginWithdrawalTimeoutUpdate(uint256 newWithdrawalTimeout)
+        external
+        onlyOwner
+    {
+        assetPool.beginWithdrawalTimeoutUpdate(newWithdrawalTimeout);
+    }
+
+    function finalizeWithdrawalTimeoutUpdate() external onlyOwner {
+        assetPool.finalizeWithdrawalTimeoutUpdate();
+    }
+
+    /// @notice Returns the time remaining until the risk manager approval
     ///         process can be finalized
-    /// @param riskManager Risk manager in the process of unapproval
+    /// @param riskManager Risk manager in the process of approval
     /// @return Remaining time in seconds.
-    function getRemainingRiskManagerUnapprovalTime(address riskManager)
+    function getRemainingRiskManagerApprovalTime(address riskManager)
         external
         view
         returns (uint256)
     {
         return
             getRemainingChangeTime(
-                riskManagerUnapprovalTimestamps[riskManager],
-                CoveragePoolConstants.RISK_MANAGER_GOVERNANCE_DELAY,
-                "Risk manager unapproval not initiated"
+                riskManagerApprovalTimestamps[riskManager],
+                assetPool.withdrawalGovernanceDelay()
             );
     }
 
     /// @notice Get the time remaining until the function parameter timer
     ///         value can be updated.
     /// @param changeTimestamp Timestamp indicating the beginning of the change.
     /// @param delay Governance delay.
-    /// @param errorMsg Revert message when change not initiated
     /// @return Remaining time in seconds.
-    function getRemainingChangeTime(
-        uint256 changeTimestamp,
-        uint256 delay,
-        string memory errorMsg
-    ) internal view returns (uint256) {
-        require(changeTimestamp > 0, errorMsg);
+    function getRemainingChangeTime(uint256 changeTimestamp, uint256 delay)
```

Comment:
This function looks almost identical to the one in asset pool. Thoughts on extracting it to a lib maybe or call `assetPool.getRemainingChangeTime()`?

Reply:
Good point, done in https://github.com/keep-network/coverage-pools/pull/105/commits/137849942d8ce99c5ffcece349aaec70bf039899.